modules/fluid_properties/src/userobjects/Water97FluidProperties.C

```diff
@@ -0,0 +1,1411 @@
+      subregion = 23;
+  }
+  else if (pMPa > 22.064 && pMPa <= 22.11)
+  {
+    if (temperature > tempXY(pressure, "qu") && temperature <= tempXY(pressure, "uv"))
+      subregion = 20;
+    else if (temperature > tempXY(pressure, "uv") && temperature <= tempXY(pressure, "ef"))
+      subregion = 24;
+    else if (temperature > tempXY(pressure, "ef") && temperature <= tempXY(pressure, "wx"))
+      subregion = 25;
+    else // (temperature > tempXY(pressure, "wx") && temperature <= tempXY(pressure, "rx"))
+      subregion = 23;
+  }
+  else
+    mooseError("Water97FluidProperties::subregion3. Shouldn't have got here!");
+
+  return subregion;
+}
+
+Real
+Water97FluidProperties::tempXY(Real pressure, std::string xy) const
+{
+  Real pi = pressure / 1.0e6;
+
+  const std::vector<std::vector<int> > I {{0, 1, 2, -1, -2}, {0, 1, 2, 3}, {0, 1, 2, 3, 4}, {0, 1, 2, 3, 4}, {0, 1, 2, 3, 4},
+                       {0, 1, 2, 3}, {0, 1, 2, -1, -2}, {0, 1, 2, 3}, {0, 1, 2, 3}, {0, 1, 2, 3, 4}, {0, 1, 2, -1, -2}};
+
+  const std::vector<std::vector<Real> > n {{0.154793642129415e4, -0.187661219490113e3, 0.213144632222113e2, -0.191887498864292e4, 0.918419702359447e3},
+                        {0.585276966696349e3, 0.278233532206915e1, -0.127283549295878e-1, 0.159090746562729e-3},
+                        {-0.249284240900418e5, 0.428143584791546e4, -0.269029173140130e3, 0.751608051114157e1, -0.787105249910383e-1},
+                        {0.584814781649163e3, -0.616179320924617, 0.260763050899562, -0.587071076864459e-2, 0.515308185433082e-4},
+                        {0.617229772068439e3, -0.770600270141675e1, 0.697072596851896, -0.157391839848015e-1, 0.137897492684194e-3},
+                        {0.535339483742384e3, 0.761978122720128e1, -0.158365725441648, 0.192871054508108e-2},
+                        {0.969461372400213e3, -0.332500170441278e3, 0.642859598466067e2, 0.773845935768222e3, -0.152313732937084e4},
+                        {0.565603648239126e3, 0.529062258221222e1, -0.102020639611016, 0.122240301070145e-2},
+                        {0.584561202520006e3, -0.102961025163669e1, 0.243293362700452, -0.294905044740799e-2},
+                        {0.528199646263062e3, 0.890579602135307e1, -0.222814134903755, 0.286791682263697e-2},
+                        {0.728052609145380e1, 0.973505869861952e2, 0.147370491183191e2, 0.329196213998375e3, 0.873371668682417e3}};
+
+  // Choose the constants based on the string xy
+  unsigned int row;
+
+  if (xy == "ab")
+    row = 0;
+  else if (xy == "cd")
+    row = 1;
+  else if (xy == "gh")
+    row = 2;
+  else if (xy == "ij")
+    row = 3;
+  else if (xy == "jk")
+    row = 4;
+  else if (xy == "mn")
+    row = 5;
+  else if (xy == "op")
+    row = 6;
+  else if (xy == "qu")
+    row = 7;
+  else if (xy == "rx")
+    row = 8;
+  else if (xy == "uv")
+    row = 9;
+  else if (xy == "wx")
+    row = 10;
+  else if (xy == "ef")
+    row = 0; // not used
```

Comment:
I see now, "ef" does not require `row`... OK. Maybe not have that in the if-statement then?